pkg/cmd/cli/cmd/set/env.go

```diff
@@ -48,6 +48,12 @@ syntax.`
   # Update all containers in all replication controllers in the project to have ENV=prod
   %[1]s env rc --all ENV=prod
 
+  # Import environment from a secret
+  %[1]s env --from-secret=mysecret dc/myapp
```

Comment:
there's nothing that prevents the client from still having resource-specific intelligence even if the syntax is
--from=secret/mysecret, since the client still knows the resource type.  so i would also vote for that approach, unless we think we're also going to be introducing additional arguments that only make sense for one type or another (at which point yes, saying "you can't use --from-secret with --configflag" is nicer than "you can't use --from with --configflag because you passed a secret resource type")